src/Bicep.Core.UnitTests/Diagnostics/LinterRuleTests/LinterRuleTestsBase.cs

```diff
@@ -16,21 +16,49 @@ namespace Bicep.Core.UnitTests.Diagnostics.LinterRuleTests
 {
```

Comment:
Changed all linter rule tests so that they fail if the code being tested has compiler errors, unless specifically intended.

Reply:
If needed I could break this out into another PR